pkg/scheduler/scheduler.go

```diff
@@ -221,14 +229,23 @@ func (s *Scheduler) Run(ctx context.Context) {
 	// Starting the scheduling queue.
 	s.queue.Run()
 
-	// Run scheduleOnce forever.
-	//
-	// The loop starts in a dedicated goroutine; it exits when the context is canceled.
-	go wait.UntilWithContext(ctx, s.scheduleOnce, 0)
+	wg := &sync.WaitGroup{}
+	wg.Add(s.workerNumber)
+	for i := 0; i < s.workerNumber; i++ {
+		go func() {
+			defer wg.Done()
+			defer utilruntime.HandleCrash()
+			// Run scheduleOnce forever.
+			//revive:disable:empty-block
+			for s.scheduleOnce(ctx) {
```

Comment:
Hi Ryan! Just out of curiosity, is there any particular reason why 

```
for {
    scheduleOnce(ctx)
}
```

does not work here?

Reply:
Also, the returned bool does not seem to be used at all; is it for future uses?

Reply:
> Hi Ryan! Just out of curiosity, is there any particular reason why
> 
> ```
> for {
>     scheduleOnce(ctx)
> }
> ```
> 
> does not work here?

how will this for loop end?

Reply:
> Also, the returned bool does not seem to be used at all; is it for future uses?

the return bool is used so that the for loop will end if the ctx is done.

Reply:
Oh, not sure what I was thinking before 😣 Though, in this case, just a nit: would be it more idiomatic to use

```
for {
    select {
    case <-ctx.Done():
        return
    default:
        scheduleOnce(ctx)
    }
}
```